tests/main/interfaces-polkit/task.yaml

```diff
@@ -0,0 +1,52 @@
+summary: Ensure that the polkit interface works.
+
+systems: [-ubuntu-core-*]
+
+prepare: |
+    if ! tests.session has-session-systemd-and-dbus; then
+        exit 0
+    fi
+    tests.session -u test prepare
+
+restore: |
+    if ! tests.session has-session-systemd-and-dbus; then
+        exit 0
+    fi
+    rm -f /home/test/sleep.stamp
+    tests.session -u test restore
+
+execute: |
+    # We don't actually need a D-Bus session bus, but this is a good
+    # guard for support for a user session.
+    if ! tests.session has-session-systemd-and-dbus; then
+        exit 0
+    fi
+
+    echo "Install the test snap"
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-pk-service
+
+    echo "The polkit interface is disconnected by default"
+    snap connections test-snapd-pk-service | MATCH "polkit +test-snapd-pk-service:polkit +- +-"
+
+    echo "The polkit policy file is not installed"
+    test ! -f /usr/share/polkit-1/actions/snap.test-snapd-pk-service.interface.polkit.foo.policy
+
+    echo "The interface can be connected"
+    snap connect test-snapd-pk-service:polkit
+    snap connections test-snapd-pk-service | MATCH "polkit +test-snapd-pk-service:polkit +:polkit +manual"
+
+    echo "Snapd has installed the policy file for the service"
+    test -f /usr/share/polkit-1/actions/snap.test-snapd-pk-service.interface.polkit.foo.policy
+    echo "The contents match the file provided by the snap"
+    cmp /usr/share/polkit-1/actions/snap.test-snapd-pk-service.interface.polkit.foo.policy ./test-snapd-pk-service/meta/polkit/polkit.foo.policy
```

Comment:
FWIW `diff` also returns 0 when files are identical but also provides a nice diff of the contents